### PR TITLE
[V3] Report Tool

### PR DESCRIPTION
**Pull request by mikeshardmind**
### Type

- [ ] Bugfix
- [ ] Enhancement
- [X] New feature

### Description of the changes
See #1280 

This has not been tested at all, and has more to be done. Not even ready for testing yet.

### CODE REVIEW

File: redbot/cogs/reports/reports.py [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+    def __init__(self, bot: Red):
+        self.bot = bot
+        self.settings = Config.get_conf(
+            self, 78631113035100160, force_registration=True)
+        self.settings.register_guild(**self.default_guild_settings)
+        self.settings.register_global(**self.default_global_settings)
+        self.antispam = {}
+        self.user_cache = []
+
+    @checks.admin_or_permissions(manage_server=True)
```
Comment (palmtree5) [inline redbot/cogs/reports/reports.py]: `manage_server` -> `manage_guild`

File: redbot/cogs/reports/reports.py [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+    @commands.group(name="reportset")
+    async def reportset(self, ctx: RedContext):
+        """
+        settings for reports
+        """
+        if ctx.invoked_subcommand is None:
+            return await ctx.send_help()
+
+    @checks.admin_or_permissions(Manage_server=True)
+    @reportset.command(name="output", pass_context=True, no_pm=True)
```
Comment (palmtree5) [inline redbot/cogs/reports/reports.py]: `pass_context=True` isn't necessary
`no_pm=True` should be removed and `@commands.guild_only()` should be put on a new line

**Review by palmtree5 — DISMISSED** [review]
I realize this is WIP but I just wanted to point out a couple issues I immediately noticed. Some occur in multiple places as well.

**Comment by mikeshardmind** [comment]
Yeah, As this is largely just a modification of my existing cog and folding it into core red, I've been avoiding completely rewriting functions unless the function actually needs rewriting., and yes, very WIP.

**Comment by Kowlin** [comment]
If this is about to get added, I do recommend you add your self to ``.github/CODEOWNERS``

**Comment by mikeshardmind** [comment]
the moderator communication portion of this is largely untested. Handling this later, but this won't be done this week at this rate.

**Comment by laggron42** [comment]
didn't see that issue; I made a report tool, I can give it to you if you want to make it core

**Comment by mikeshardmind** [comment]
Hey, surprising even myself, this is now tested locally and ready for review.

**Review by palmtree5 — APPROVED** [review]

**Pull request merged**